### PR TITLE
React Frontend and simple attack implementation

### PR DESCRIPTION
**Pull request by Pluppen**
Its alot of stuff but woop woop.

There will be needed some restructering later and also refactor but is good for now

### CODE REVIEW

File: frontend/src/Game.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    extraHtml: string;
+}
+
+
+const Game: React.FC = () => {
+    const [accounts, setAccounts] = useState<IAccount[]>([]);
+    const [modalInfo, setModalInfo] = useState<IModalInfo | null>(null);
+
+    useEffect(() => {
+        fetch("http://localhost:3000/account").then(res => res.json()).then((res) => {
```
Comment (Alexamakans) [inline frontend/src/Game.tsx:29]: Let's make `http://localhost:3000` some config thing from the start to make it easier on us later imo, could be just pull from env var idc tbh

```suggestion
        fetch(`${config.BASE_URL}/account`).then(res => res.json()).then((res) => {
```

Maybe with the format `HEALTYBOICHALLENGE_[varname]`
so
`HEALTYBOICHALLENGE_BASE_URL` or something

or if there's something neat built into react

File: routes/index.js [hunk 2]
```diff
@@ -1,142 +1,166 @@
...
-                return obj;
-            }),
-        ),
+    res.json(
+        game.accounts.map((a) => {
+            const obj = { ...a };
+            obj.passwordHash = "";
+            obj.passwordSalt = "";
+            return obj;
+        }),
```
Comment (Alexamakans) [inline routes/index.js:67]: this thing you did was nicer

```suggestion
        game.accounts.map((a) => ({
            ...a,
            passwordHash: "",
            passwordSalt: "",
        })),
```

File: routes/index.js [hunk 3]
```diff
@@ -1,142 +1,166 @@
...
+    account.vertexShaderCode = req.body.vertexShaderCode;
+    account.fragmentShaderCode = req.body.fragmentShaderCode;
+    account.useShaderForProfilePicture = req.body.useShaderForProfilePicture;
 
     game.save();
-    res.redirect("/");
+    res.json({
+        success: true,
+        message: "Updated shader profile",
+        account: { ...account, passwordHash: "", passwordSalt: "" },
```
Comment (Alexamakans) [inline routes/index.js:154]: can we middleware this thing somehow, or maybe we should split it so account is like
account.data = the stuff we have except password
account.cred = password stuff
and just send account.data?

wanna get rid of having to manipulate the accounts array each time we send all accounts though, any idea for separating it?

out of scope for this PR of course

**Review by Alexamakans — APPROVED** [review]
LGTM

minor comments

File: routes/index.js (hunk 3 above)
Comment (Pluppen) [reply routes/index.js:153]: We should probably be able to do this in a nice way, maybe without having to change the underlying structure too much and without middleware. It feels like there should be a way to make the sensitive fields more of a optin when getting data.

Comment (Pluppen) [reply routes/index.js:153]: https://github.com/Pluppen/HealtyBoiChallenge/issues/17

File: frontend/src/Game.tsx (hunk 1 above)
Comment (Pluppen) [reply frontend/src/Game.tsx]: Fixed by using environment variable. By creating a .env file in the root of the frontend folder containing the REACT_APP_BASE_URL="url" React will load it into environment variables making it usable using process.env

**Pull request merged**